src/app/login/login.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../auth.service';
+import { Router } from '@angular/router';
+import { FormControl, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+  username = new FormControl();
+  password = new FormControl();
+
+  constructor(private Auth: AuthService,
+              private router: Router) { }
+
+  ngOnInit() {
+  }
+
+  loginUser(event) {
+    event.preventDefault();
+    const target = event.target;
+    const username = target.querySelector('#username').value;
+    const password = target.querySelector('#password').value;
+    console.log(username, password);
+
+    this.Auth.getUserDetails(username, password).subscribe(data => {
+      if (data.success) {
+        this.router.navigate(['home']);
+        this.Auth.setLoggedIn(true);
```

Comment:
You might consider switching these around. You first navigate to other component, and then call on Auth service, which is injected into this component? Switch 30 and 31 around.